src/Service/Core.php

```diff
@@ -156,28 +131,18 @@ public function getAppConfig($module, $language) {
    *   The machine name of the extending module.
    * @param string $language
    *   The language code.
-   *
-   * @return \Drupal\Core\Entity\EntityInterface|null
-   *   The relevant applanding node.
-   *
-   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
-   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
    */
   public function getAppNode($module, $language) {
-    // TODO: dependency injection.
-    // TODO: prevent more than one node per language being created for each app.
-    // TODO: filter by language.
-    $query = \Drupal::entityQuery('node')
+    $node_details = [];
+    $nids = \Drupal::entityQuery('node')
```

Comment:
This should be done using DI using services.yml 
https://www.drupal.org/docs/8/api/services-and-dependency-injection/dependency-injection-for-a-form

Reply:
@dakkusingh , @malcomio : The current implementing is working as required. As commented by Janak, I tried implementing the DI concept but getting some error related to service. I am working on that. But till this process can be implemented.